### PR TITLE
The Numnum Bug

### PR DESCRIPTION
**Pull request by yhahn**
Ran into this debugging a wacky address query in Sweden:

```
115 39 Sweden
```

Can currently return something like 

```
# this has relevance 1
115 115, 269 76 Torekov, Skåne län, Sweden
```

What's going on here where we end up with the house 115 on street 115 but no sign of 39 to be found?

### What's going on

- Query gets numtokenized and normalized to these two permutations: `## 115` `1## 39` (we always index the data with housenum first).
- `## 115` matches data
- On the way back out, the `maskAddress` is `11` (both term positions are potential house numbers) and maskAddress is called against the original _query_ which has tokens in this order `115 39`
- It snags the first housenum-like value, `115`
- And pairs it with the street that's been found, `115`

And so we end up with `115 115`.

### The short-term fix (in this branch)

Make `maskAddress` a bit smarter by looking at both the coverText and query to determine if it's about to reuse a housenum that was really originally interpreted as a street.

### The long-term fix

When we `numTokenize` at the initial phrasematch stage, extracting the housenumber early may make more sense rather than masking and then doing forensics later to try to guess at what happened.

---

For now have put the stopgap short term fix in place.

cc @mapbox/geocoding-gang

### CODE REVIEW

File: lib/util/termops.js [hunk 1]
```diff
@@ -151,14 +151,20 @@ function tokenize(query, lonlat) {
  * is a valid address
  *
  * @param {Array} query tokenized query
+ * @param {Array} tokenized cover text
```
Comment (mattficke) [inline lib/util/termops.js]: I think this needs the `coverText` parameter name added?

File: lib/util/termops.js [hunk 2]
```diff
@@ -151,14 +151,20 @@ function tokenize(query, lonlat) {
...
  * @param {Integer} mask a mask for the given query
  * @return {Object} returns an address object or null
 */
-function maskAddress(query, mask) {
+function maskAddress(query, coverText, mask) {
+    var coverTokens = new Set(tokenize(coverText));
```
Comment (mattficke) [inline lib/util/termops.js]: Ooh. I haven't used Set before, this is a nice example.

File: lib/util/termops.js (hunk 1 above)
Comment (yhahn) [reply lib/util/termops.js]: :+1:

File: lib/util/termops.js (hunk 2 above)
Comment (yhahn) [reply lib/util/termops.js]: learned from @apendleton

**Pull request merged**